explorer-api/src/geo_ip/location.rs

```diff
@@ -42,6 +46,17 @@ pub(crate) struct Location {
     pub(crate) longitude: Option<f64>,
 }
 
+pub(crate) struct FailedIpAddresses {
+    failed_ips: Arc<Mutex<HashSet<String>>>,
+}
+
+impl FailedIpAddresses {
+    pub fn new() -> Self {
+        FailedIpAddresses {
+            failed_ips: Arc::new(Mutex::new(HashSet::new())),
```

Comment:
You wan't one more thing here -> on startup it should check if the file already exists and load its content because on each restart you'll end up duplicating everything

Reply:
Also, does it need to be in a Arc Mutex? the parent struct is already in a Arc?

Reply:
oh yeah, probably you don't need the `Arc` at all

Reply:
give me a second to make changes guys 😉